### PR TITLE
Creating all needed templates addRemoveQuestions.html createSurvey.ht…

### PR DESCRIPTION
**Pull request by MiniEinsteinD**
Creating all needed templates addRemoveQuestions.html createSurvey.html error.html surveyCreated.html surveyResults.html
 and fixed the ControllerStructure.java

### CODE REVIEW

File: src/main/java/org/group23/ControllerStructure.java [hunk 1]
```diff
@@ -1,55 +1,108 @@
...
-        model.addAttribute("message", "Hello, World!");
-        return "helloPage";
+    @GetMapping("/saveSurvey")
+    public String saveSurvey() {
+        return "redirect:/surveyCreated";
+    }
+
+    @GetMapping("/surveyCreated")
+    public String surveyCreated(Model model) {
+        Survey survey = surveyRepository.findById(1L).orElse(null);
```
Comment (Andrei486) [inline src/main/java/org/group23/ControllerStructure.java:97]: This should try to get whichever survey was created, not the one with ID 1.

File: src/main/resources/templates/surveyCreated.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+<html xmlns="http://www.w3.org/1999/xhtml"
+      xmlns:th="http://www.thymeleaf.org">
+<head>
+  <meta charset="UTF-8">
+  <title>Survey Created</title>
+</head>
+<body>
+<h2>Survey Created Successfully</h2>
+<p>The survey with name <span th:text="${survey.name}"></span> has been created.</p>
+<a th:href="@{/createSurvey}" class="btn btn-primary">Create Another Survey</a>
```
Comment (Andrei486) [inline src/main/resources/templates/surveyCreated.html:11]: This should also have a link to add/remove questions from the newly created survey.

File: src/main/resources/templates/addRemoveQuestions.html [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+<!-- Display existing questions and allow deletion -->
+<ul>
+    <li th:each="question : ${survey.questions}">
+        <span th:text="${question.question}"></span>
+        <a th:href="@{'/deleteQuestion/' + ${survey.id} + '/' + ${question.id}}">Delete</a>
+    </li>
+</ul>
+<!-- Add other content or actions as needed -->
+
+<a th:href="@{/saveSurvey}" class="btn btn-primary">Save Survey</a>
```
Comment (Andrei486) [inline src/main/resources/templates/addRemoveQuestions.html:24]: No point to this button, adding and deleting questions already saves the changes.

**Review by Andrei486 — CHANGES_REQUESTED** [review]
Checked on my local and the web app seems to work. I won't approve this to merge until you talk with @burg1337 and @Rayatk who were supposed to do the views in the first place.

File: src/main/java/org/group23/ControllerStructure.java [hunk 4]
```diff
@@ -1,55 +1,108 @@
...
+            // Handle survey not found
+            return "error";
+        }
+    }
 
+    @PostMapping("/addQuestion/{surveyId}")
+    public String addQuestion(
+            @PathVariable Long surveyId,
+            @ModelAttribute("survey") Survey survey,
+            @RequestParam String questionText,
```
Comment (Andrei486) [inline src/main/java/org/group23/ControllerStructure.java:45]: Validate length before adding to database!

Comment (Andrei486) [reply src/main/java/org/group23/ControllerStructure.java:45]: image

File: src/main/java/org/group23/ControllerStructure.java (hunk 1 above)
Comment (MiniEinsteinD) [reply src/main/java/org/group23/ControllerStructure.java:97]: done

File: src/main/java/org/group23/ControllerStructure.java (hunk 4 above)
Comment (MiniEinsteinD) [reply src/main/java/org/group23/ControllerStructure.java:45]: done

**Review by Andrei486 — APPROVED** [review]
Looks good, tests are in @burg1337 's PR so that's good.

**Pull request merged**